lighthouse-core/audits/dobetterweb/uses-optimized-images.js

```diff
@@ -0,0 +1,141 @@
+      helpText: `
+        Images should be optimized to save network bytes.
+        The following images could have smaller file sizes when compressed with
+        [WebP](https://developers.google.com/speed/webp/) or JPEG at 80 quality.
+        [Learn more about image optimization](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/image-optimization).
+      `.trim(),
+      requiredArtifacts: ['OptimizedImages']
+    };
+  }
+
+  /**
+   * @param {{originalSize: number, webpSize: number, jpegSize: number}} image
+   * @param {string} type
+   * @return {{bytes: number, kb: number, percent: number}}
+   */
+  static computeSavings(image, type) {
+    const bytes = image.originalSize - image[type + 'Size'];
+    const kb = Math.round(bytes / KB_IN_BYTES);
+    const percent = Math.round(100 * bytes / image.originalSize);
+    return {bytes, kb, percent};
+  }
+
+  /**
+   * @param {{isBase64DataUri: boolean, url: string}} image
+   * @return {string}
+   */
+  static getUrl(image) {
+    return image.isBase64DataUri ? image.url.slice(0, 80) + '...' : image.url;
+  }
+
+  /**
+   * @param {!Artifacts} artifacts
+   * @return {!AuditResult}
+   */
+  static audit(artifacts) {
+    const images = artifacts.OptimizedImages;
+
+    if (images.rawValue === -1) {
+      return UsesOptimizedImages.generateAuditResult(images);
+    }
+
+    const failedImages = [];
+    let totalWastedBytes = 0;
+    let hasAllEfficientImages = true;
+
+    const results = images.reduce((results, image) => {
+      if (image.failed) {
+        failedImages.push(image);
+        return results;
+      } else if (image.originalSize < Math.max(IGNORE_THRESHOLD_IN_BYTES, image.webpSize)) {
+        return results;
+      }
+
+      const originalKb = Math.round(image.originalSize / KB_IN_BYTES);
+      const url = UsesOptimizedImages.getUrl(image);
+      const webpSavings = UsesOptimizedImages.computeSavings(image, 'webp');
+
+      let label = `${originalKb} KB total, webp savings: ${webpSavings.percent}%`;
+      if (webpSavings.bytes > WEBP_ALREADY_OPTIMIZED_THRESHOLD_IN_BYTES) {
+        hasAllEfficientImages = false;
+      }
+
+      if (/(jpeg|bmp)/.test(image.mimeType)) {
+        const jpegSavings = UsesOptimizedImages.computeSavings(image, 'jpeg');
+        if (jpegSavings.bytes > 0) {
+          hasAllEfficientImages = false;
+          label += `, jpeg savings: ${jpegSavings.percent}%`;
+        }
+      }
+
+      totalWastedBytes += webpSavings.bytes;
+      results.push({url, label});
+      return results;
+    }, []);
+
+    let displayValue = '';
+    if (totalWastedBytes > 1000) {
+      displayValue = `${Math.round(totalWastedBytes / KB_IN_BYTES)}KB potential savings`;
+    }
+
+    let debugString;
+    if (failedImages.length) {
+      const urls = failedImages.map(image => UsesOptimizedImages.getUrl(image));
+      debugString = `Lighthouse was unable to parse some of your images: ${urls.join(', ')}`;
```

Comment:
No it's usually canvas failing because the image was bad/not something we could read. Decode probably makes more sense here?